sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -93,13 +95,22 @@ object IntervalUtils {
 
   private val yearMonthPattern = "^([+|-])?(\\d+)-(\\d+)$".r
 
+  def fromYearMonthString(input: UTF8String): CalendarInterval = {
+    if (input == null || input.toString == null) {
+      throw new IllegalArgumentException("Interval year-month string must be not null")
+    } else {
+      fromYearMonthString(input.trimAll().toString)
+    }
+  }
+
   /**
    * Parse YearMonth string in form: [+|-]YYYY-MM
    *
    * adapted from HiveIntervalYearMonth.valueOf
    */
   def fromYearMonthString(input: String): CalendarInterval = {
```

Comment:
Now I understand why we need a new version of `fromYearMonthString`, as we need to allow `INTERVAL ... YEAR TO MONTH`. Let's make it clear in the PR description.

Let's add a new method `castStringToYMInterval`, and define a new regex for it.

Reply:
Done， how about current?